agent/agent.go

```diff
@@ -2900,6 +2906,45 @@ func (a *Agent) addCheck(check *structs.HealthCheck, chkType *structs.CheckType,
 			dockerCheck.Start()
 			a.checkDockers[cid] = dockerCheck
 
+		case chkType.IsOSService():
+			if existing, ok := a.checkOSServices[cid]; ok {
+				existing.Stop()
+				delete(a.checkOSServices, cid)
+			}
+			if chkType.Interval < checks.MinInterval {
+				a.logger.Warn("check has interval below minimum",
+					"check", cid.String(),
+					"minimum_interval", checks.MinInterval,
+				)
+				chkType.Interval = checks.MinInterval
+			}
+
+			if a.osServiceClient == nil {
+				ossp, err := checks.NewOSServiceClient()
+				if err != nil {
+					a.logger.Error("error creating OS Service client", "error", err)
+					return err
+				}
+				a.logger.Debug("created OS Service client")
+				a.osServiceClient = ossp
+			}
+
+			osServiceCheck := &checks.CheckOSService{
+				CheckID:       cid,
+				ServiceID:     sid,
+				OSService:     chkType.OSService,
+				Timeout:       chkType.Timeout,
+				Interval:      chkType.Interval,
+				Logger:        a.logger,
+				Client:        a.osServiceClient,
+				StatusHandler: statusHandler,
+			}
+			if prev := a.checkOSServices[cid]; prev != nil {
+				prev.Stop()
+			}
```

Comment:
@deblasis, @kisunji : what's the difference between this condition and the one on lines L2910-L2913?
```
			if existing, ok := a.checkOSServices[cid]; ok {
				existing.Stop()
				delete(a.checkOSServices, cid)
			}
```

It seems like both cases involve checking to see if a previous check exists, and if it does, stopping it. But the first invocation also has a delete operation.

Am I misreading something?

Reply:
Good eye! I think this may be a copy-paste from the docker check case above. This block looks unnecessary:
```go
if prev := a.checkOSServices[cid]; prev != nil {
	prev.Stop()
}
```

Reply:
You are right guys, it's a copy-paste from the Docker check above. 

I guess we should log that we have to remove the redundant lines in the Docker check as well.
Unless you want to do it in this PR

Reply:
I wouldn't mind if you just removed the line from the docker case as well